libraries/chain/asset.cpp

```diff
@@ -80,13 +77,42 @@ asset asset::from_string(const string& from)
       return result;
    }
    FC_CAPTURE_LOG_AND_RETHROW( (from) )
+   */
+   try { 
+      string s = fc::trim( from );
+      auto space_pos = s.find( " " );
+      auto dot_pos = s.find( "." );
+      
+      asset result;
+      
+      auto intpart = s.substr( 0, dot_pos );
+      result.amount = fc::to_int64(intpart);
```

Comment:
This fails for the case of "10 CUR". Seems like that should be allowed. If not, then add a better error than couldn't parse int64_t.  Also add test cases for this.

Reply:
Added more checks to both asset and symbol classes. If precision is now missing, an exception is throw.